portfolio/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -0,0 +1,67 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.User;
+import com.google.gson.Gson;
+
+@WebServlet("/user")
+public class UserServlet extends HttpServlet {
+    /**
+     * Checks state of user in this class.
+     */
+    private static UserService userService;
+
+    /**
+     * Used to convert Java objects into JSON
+     */
+    private static Gson gson;
+    @Override
+    public void init() {
+        userService = UserServiceFactory.getUserService();
+        gson = new Gson();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String url = "";
```

Comment:
nit: Java compiler should be smart enough to know this has to be initialized, without having to initialize to empty string here, since it is set in the if else control flow below.